### PR TITLE
Moved all message handling into the new interface.

### PR DESCRIPTION
**Pull request by davidv1992**

### CODE REVIEW

File: statime/src/port/tests.rs [hunk 1]
```diff
@@ -1,166 +0,0 @@
...
-    }
-}
-
-#[derive(Debug, Clone, Default, PartialEq, Eq)]
-pub struct TestNetworkPacket {
-    data: ArrayVec<u8, MAX_DATA_LEN>,
-}
-
-#[tokio::test]
-async fn test_measurement_flow() {
```
Comment (davidv1992) [inline statime/src/port/tests.rs:89]: Test is duplicate with the tests in state/slave.rs

**Review by bramtweedegolf — APPROVED** [review]

File: statime/src/port/mod.rs [hunk 2]
```diff
@@ -70,61 +77,153 @@ pub enum PortAction<'a> {
...
+            self.config.port_identity,
+            default_ds,
+            buffer,
+        ));
+
+        if let Err(error) =
+            self.temp_handle_time_measurement(local_clock, filter, time_properties_ds)
+        {
+            log::error!("Could not handle time measurement: {:?}", error);
+        }
```
Comment (folkertdev) [inline statime/src/port/mod.rs:172]: this does not (need to) influence what is returned?

File: statime/src/port/state/slave.rs [hunk 3]
```diff
@@ -179,20 +218,27 @@ impl SlaveState {
...
+                .serialize(buffer)
+                .map_err(|error| {
+                    log::error!("Could not serialize delay request: {:?}", error);
+                    error
+                })
+                .ok()?;
             self.delay_state = DelayState::Measuring {
                 id: delay_id,
-                send_time: Some(delay_send_time),
+                send_time: None,
```
Comment (folkertdev) [inline statime/src/port/state/slave.rs:230]: just checking: is this correct?

**Review by folkertdev — APPROVED** [review]
left 2 questions. I suspect they are not blockers

**Pull request merged**